src/actions.ml

```diff
@@ -444,26 +444,30 @@ let job_action ~bot_info ({build_name} as job_info) ~gitlab_mapping =
   | unknown_state ->
       Lwt_io.printf "Unknown job status: %s\n" unknown_state
 
-let create_pipeline_summary ?summary_top pipeline_info pipeline_url =
+let create_pipeline_summary ?(summary_top = "") pipeline_info pipeline_url =
   let sorted_builds =
     pipeline_info.builds
     |> List.sort ~compare:(fun build1 build2 ->
            String.compare build1.build_name build2.build_name )
   in
-  pipeline_info.stages
-  |> List.concat_map ~f:(fun stage ->
-         sorted_builds
-         |> List.filter_map ~f:(fun build ->
-                if String.equal build.stage stage then
-                  Some
-                    (f "  - [%s](%s/-/jobs/%d)" build.build_name
-                       pipeline_info.common_info.repo_url build.build_id )
-                else None )
-         |> List.cons ("- " ^ stage) )
-  |> List.cons
-       (f "This [GitLab pipeline](%s) contains the following stages and jobs:"
-          pipeline_url )
-  |> (match summary_top with Some text -> List.cons text | None -> Fn.id)
+  let stage_summary =
+    pipeline_info.stages
+    |> List.concat_map ~f:(fun stage ->
+           sorted_builds
+           |> List.filter_map ~f:(fun build ->
+                  if String.equal build.stage stage then
+                    Some
+                      (f "  - [%s](%s/-/jobs/%d)" build.build_name
+                         pipeline_info.common_info.repo_url build.build_id )
+                  else None )
+           |> List.cons ("- " ^ stage) )
+    |> String.concat ~sep:"\n\n"
```

Comment:
We can also decrease this to a single new line to stop the summary from being so spaced out.

Reply:
Sounds like a good idea.